### PR TITLE
Fixed code coverage for crypto tests

### PR DESCRIPTION
**Pull request by justanotherminh**
`SYSCALL_NO_INLINE` is removed in musl version 1.2.2, causing code coverage to fail for tests under `tests/crypto`. This PR fixes this issue.

### CODE REVIEW

**Review by mingweishih — APPROVED** [review]

**Comment by mingweishih** [comment]
bors try

**Review by dcarpente — APPROVED** [review]

File: tests/crypto/enclave/enc/enc.c [hunk 1]
```diff
@@ -145,64 +147,25 @@ OE_DEFINE_SYSCALL1_M(SYS_close)
...
-
-    if (!ret)
-        OE_RAISE(OE_INVALID_PARAMETER);
+#if !defined(CODE_COVERAGE) && defined(OE_USE_OPENSSL)
+    /*
+     * When enabling code coverage analysis, libgcov should initialize the host
+     * fs already so we do not do it again here. Otherwise, the
+     * oe_load_module_host_file_system will fail.
+     */
+    if (oe_load_module_host_file_system() != OE_OK)
```
Comment (radhikaj) [inline tests/crypto/enclave/enc/enc.c:160]: @mingweishih  @justanotherminh is this correct? If CODE_COVERAGE is not defined, the file system module is loaded. It should be if CODE_COVERAGE is defined.

Comment (mingweishih) [reply tests/crypto/enclave/enc/enc.c:160]: This is correct. When CODE_COVERAGE is enabled, the `libgcov` will do this for us. When CODE_COVERAGE is not enabled, we will need to make this call to opt into the host fs.

**Review by radhikaj — COMMENTED** [review]
@justanotherminh  Can you confirm the code works correctly when code coverage is enabled and when it is not? Also what about openssl=openssl_3

**Comment by radhikaj** [comment]
bors try

File: tests/crypto/enclave/enc/enc.c [hunk 2]
```diff
@@ -41,6 +42,7 @@ char* oe_host_strdup(const char* str)
 
 // Tests need these syscall overrides.
 
+#if !defined(CODE_COVERAGE)
```
Comment (mingweishih) [inline tests/crypto/enclave/enc/enc.c:45]:
```suggestion
#if !defined(OE_USE_OPENSSL)
```

**Comment by justanotherminh** [comment]
> @justanotherminh Can you confirm the code works correctly when code coverage is enabled and when it is not? Also what about openssl=openssl_3

I've tried it with both code coverage toggled on and off and it was OK. The flag OE_USE_OPENSSL is defined for both versions of openssl so that should be fine as well.

**Comment by justanotherminh** [comment]
bors try

File: tests/crypto/enclave/enc/CMakeLists.txt [hunk 3]
```diff
@@ -49,6 +49,8 @@ endif ()
 enclave_include_directories(crypto_mbedtls_enc PRIVATE
                             ${CMAKE_CURRENT_BINARY_DIR})
 
+enclave_link_libraries(crypto_mbedtls_enc oehostfs)
```
Comment (mingweishih) [inline tests/crypto/enclave/enc/CMakeLists.txt:52]: This is not needed

**Comment by justanotherminh** [comment]
bors try

**Review by asvrada — APPROVED** [review]

**Comment by justanotherminh** [comment]
bors try

**Comment by justanotherminh** [comment]
bors try

**Comment by radhikaj** [comment]
Bors try

**Comment by radhikaj** [comment]
Bors try

**Comment by radhikaj** [comment]
bors try

**Pull request closed without merging**